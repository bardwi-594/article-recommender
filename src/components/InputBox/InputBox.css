form.input-form input {
    line-height: 1.2em;
    padding: 13px 24px;
    font-size: 0.875rem;
    background: #fff;
    border: 1px solid rgba(34,36,38,.15);
    color: rgba(0,0,0,.87);
    border-radius: .3rem;
    box-shadow: 0 0 0 0 transparent inset;
    transition: color .1s ease,border-color .1s ease;
}

form.input-form input:focus {
    color: rgba(0,0,0,.95);
    border-color: #1d96e8;
    border-radius: .3rem;
    outline: none;
    background: #fff;
    box-shadow: 0 0 0 0 rgb(34 36 38 / 35%) inset;
}

div.inputbox-container{
    display: flex;
    justify-content: center;
}

div.form-field {
    display: flex;
    flex-direction: row;
    margin: 0 -.5em 1em;
}

div.form-field input {
    margin-right: 0.5em;
}

div button.submit {
    background-color: #1d96e8;
    border: 0;
    display: inline-block;
    color: #fff;
    cursor: pointer;
    font-size: 0.9375rem;
    font-weight: 600;
    letter-spacing: .03em;
    padding: 13px 24px;
    border-radius: 0.3rem;
    transition: .1s ease;
}

div button.submit:focus {
    outline: none;
    opacity: 1;
}