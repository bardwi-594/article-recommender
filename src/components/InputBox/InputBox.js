import React, { Component } from "react";
import "./InputBox.css";

class InputBox extends Component {
    render() {
        return (
            <div className="inputbox-container">
                <form className="input-form">
                    <div className="form-field">
                        <input />
                        <button className="submit" type="submit" color="green">Submit</button>
                    </div>
                </form>

            </div>
        )
    }
}
export default InputBox;