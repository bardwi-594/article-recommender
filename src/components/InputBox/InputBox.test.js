import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, render, mount, configure } from 'enzyme';
import InputBox from "./InputBox";

/**Test case for InputBox Component**/

configure({ adapter: new Adapter() });

describe('Test for InputBox Component', () => {
    it('renders the InputBox component', () => {
        const component = shallow(<InputBox />);
        expect(component.exists()).toBe(true);
    });

    let wrapper;
    let mockSubmit;
    beforeEach(() => {
        mockSubmit = jest.fn();
        wrapper = shallow(<InputBox submit={mockSubmit} />);
    });

    it("should match the snapshot", () => {
        expect(wrapper).toMatchSnapshot();
    });

    //handleChange
    it("should call setState on urlText", () => {
        const mockEvent = {
            target: {
                name: "urlText",
                value: "www.google.com"
            }
        };
        const expected = {
            urlText: "www.google.com",
        };
        wrapper.instance().handleChange(mockEvent);
        expect(wrapper.state()).toEqual(expected);
    });
    /**should call handleChange on url change with the correct params**/
    it("should call handleChange on url change with the correct params", () => {
        const spy = jest.spyOn(wrapper.instance(), "handleChange");
        wrapper.instance().forceUpdate();
        const mockEvent = {
            target: {
                name: "input-text",
                value: "test"
            }
        };
        wrapper.find(".urlfield").simulate("change", mockEvent);
        expect(spy).toHaveBeenCalledWith(mockEvent);
    });
    /**when the form is submitted the event is cancelled**/
    it("when the form is submitted the event is cancelled", () => {
        const component = shallow(<InputBox inputTextBox={() => { }} />);
        let prevented = false;
        component.find("form").simulate("submit", {
            preventDefault: () => {
                prevented = true;
            }
        });
        expect(prevented).toBe(true);
    })


});