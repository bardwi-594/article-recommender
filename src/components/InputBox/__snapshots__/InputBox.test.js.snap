// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputBox Component should match the snapshot 1`] = `ShallowWrapper {}`;

exports[`Test for InputBox Component should match the snapshot 1`] = `ShallowWrapper {}`;
